src/app/user-detail/user-detail.component.ts

```diff
@@ -1,32 +1,39 @@
 import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute, Router } from '@angular/router';
+import { ActivatedRoute, Params, Router } from '@angular/router';
 import { User } from '../interfaces/user';
+import { UsersService } from '../users.service';
 
 @Component({
   selector: 'app-user-detail',
   templateUrl: './user-detail.component.html',
   styleUrls: ['./user-detail.component.scss']
 })
 export class UserDetailComponent implements OnInit {
-  user;
-  title: string;
-  detail: string;
+  user: User;
   constructor(
     private router: Router,
-    private route: ActivatedRoute
+    private route: ActivatedRoute,
+    private userService: UsersService
     ) { }
 
   ngOnInit(): void {
-    this.route.queryParams.subscribe(params => {
-      this.user = JSON.parse(params.data);
+    let userId: number;
+    this.route.params.subscribe((params:Params) => {
+      userId = +params['userId'];
     })
-    this.detail = this.user.name;
-    this.title = 'Hi, My name is';
+    if (!this.userService.newData) {
+      this.userService.getUsers().subscribe(data => {
```

Comment:
We're not using the `data` variable at all. We can just use `() =>` instead.

Reply:
done